### PR TITLE
Update to remove magic from memopol

### PR DESCRIPTION
**Pull request by jpic**
This PR contains a bunch of features extracted while removing magic and technical debt from memopol.

### CODE REVIEW

File: representatives/models.py [hunk 1]
```diff
@@ -130,6 +125,31 @@ def gender_as_str(self):
...
+
+        for m in self.mandates.all():
+            if m.end_date < today:
+                continue
+
+            if m.group.kind != 'group':
+                continue
+
+            return m
+
```
Comment (Psycojoker) [inline representatives/models.py]: Those two functions can be fully done in SQL no? like self.mandates.filter(consitituency__country_id__isnull=False)[0] and self.mandates.filter(end_date__lt=date.today(), group__kind="group")[0] or am I missing something?

Comment (jpic) [inline representatives/models.py]: If you do it in SQL, you'll do one SQL query per Representative you call. Instead, we leverage prefetch_related, implemented here (as you can see it does the select related in the prefetch to have mandate.constituency.country and group):

https://github.com/political-memory/django-representatives/pull/12/files#diff-375065bb5decf79fe0128018a8a8ecafR88

and tested here (see the assertNumQueries):

https://github.com/political-memory/django-representatives/pull/12/files#diff-77278a91b6628e42e6059a58b56a466bR19

**Comment by jpic** [comment]
<3 how the codecov browser extension show 100% coverage for this diff, yeaahhhhh

**Comment by jpic** [comment]
2015-12-22-213750_877x45_scrot
2015-12-22-213801_897x58_scrot
2015-12-22-213807_914x60_scrot
2015-12-22-213813_920x100_scrot
2015-12-22-213820_897x45_scrot
2015-12-22-213831_899x58_scrot
2015-12-22-213838_890x79_scrot

**Comment by jpic** [comment]
@Psycojoker I added a commit which I hope will make it less confusing for non-django hackers. I think it will be more to your taste even though I"ll have to rewrite all templates in memopol ;) Is this PR good to go know ?

**Comment by jpic** [comment]
Need to release this to move forward, anyway, the code @Psycojoker commented is now encapsulated in a class which makes it clear: https://github.com/political-memory/django-representatives/blob/extract_from_memopol/representatives/views.py#L10-L54

If you find anything else please open an issue and assign it to me !

**Comment by jpic** [comment]
Version 0.0.7 signed release uploaded.

File: representatives/templatetags/representatives_tags.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+def country_flag(country):
+    return mark_safe(
+        '<span class="flag-icon flag-icon-{code}"></span> {name}'.format(
+            name=country.name,
+            code=country.code.lower()))
+
+
+@register.filter
+def mandate_date(date, arg=None):
+    if date.year == 9999:
```
Comment (Psycojoker) [inline representatives/templatetags/representatives_tags.py]: I wonder if we shouldn't set this 9999 to None during important, but this is really neat picking.

File: representatives/views.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
+import datetime
+
+from django.db import models
+from django.views import generic
+from django.utils.text import slugify
+
+from .models import Mandate, Representative, Group
+
+
+class RepresentativeViewMixin(object):
```
Comment (Psycojoker) [inline representatives/views.py]: Stupid question: why a mixing and not model methods?

Comment (Psycojoker) [inline representatives/views.py]: More like a queryset method actually.

**Comment by Psycojoker** [comment]
Otherwise looks good :)

File: representatives/views.py (hunk 3 above)
Comment (jpic) [inline representatives/views.py]: It would make it more complicated to override.

File: representatives/templatetags/representatives_tags.py (hunk 2 above)
Comment (jpic) [inline representatives/templatetags/representatives_tags.py]: You can't set a datetime's year to None !

In [5]: datetime.date(year=None,month=1,day=1)
TypeError: an integer is required

I dislike this too, but can live with it.

**Pull request merged**